build_msvc.bat

```diff
@@ -1,20 +1,33 @@
 @echo off
-rem launch this from msvc-enabled console
+
+if not defined DevEnvDir (
+    if "%VSWHERE%"=="" set "VSWHERE=%ProgramFiles(x86)%\Microsoft Visual Studio\Installer\vswhere.exe"
+)
+if not defined DevEnvDir (
+    for /f "usebackq tokens=*" %%i in (`call "%VSWHERE%" -latest -products * -requires Microsoft.Component.MSBuild -property installationPath`) do (
+      set InstallDir=%%i
+    )
+)
+if not defined DevEnvDir (
+    if exist "%InstallDir%\VC\Auxiliary\Build\vcvars64.bat" (
+        call "%InstallDir%\VC\Auxiliary\Build\vcvars64.bat"
+    )
+)
```

Comment:
1. This looks like completely outside of the scope of "printing labels".

2. I'm not a windows users but this looks like a job that we should not even do. What if this does not work with some MSVC versions? Do we keep making this more and more complex until it barely works and unmaintainable? I think it's better to just require to run build_msvc.bat from the MSVC environment thing and move on. Windows is too crazy to waste our time on this kind of things.